vmware_exporter/vmware_exporter.py

```diff
@@ -369,58 +369,66 @@ def datacenter_inventory(self):
-                    row = inventory[folder.name] = [
-                        folder.name,
-                        dc.name,
-                    ]
-                    if isinstance(node, vim.StoragePod):
-                        row.append(node.name)
-                    else:
-                        row.append('')
-
-                else:  # Folder is a Datastore Cluster
-                    inventory.update(_collect(dc, folder))
+            if isinstance(node, vim.Folder) and not isinstance(node, vim.StoragePod):
+                logging.debug(f"[F ] {('-' * level).ljust(7)} {node.name}")
+                for child in node.childEntity:
+                    inventory.update(_collect(child, level+1, dc))
+            elif isinstance(node, vim.Datacenter):
+                logging.debug(f"[DC] {('-' * level).ljust(7)} {node.name}")
+                inventory.update(_collect(node.datastoreFolder, level+1, node.name))
+            elif isinstance(node, vim.Folder) and isinstance(node, vim.StoragePod):
+                logging.debug(f"[SP] {('-' * level).ljust(7)} {node.name}")
+                for child in node.childEntity:
+                    inventory.update(_collect(child, level+1, dc, node.name))
+            elif isinstance(node, vim.Datastore):
+                logging.debug(f"[DS] {('-' * level).ljust(7)} {node.name}")
+                inventory[node.name] = [node.name, dc, storagePod]
+            else:
+                logging.debug(f"[ ?] {('-' * level).ljust(7)} {node}")
             return inventory
 
         labels = {}
         dcs = yield self.datacenter_inventory
         for dc in dcs:
-            result = yield threads.deferToThread(lambda: _collect(dc, dc.datastoreFolder))
+            result = yield threads.deferToThread(lambda: _collect(dc))
             labels.update(result)
-
         return labels
 
     @run_once_property
     @defer.inlineCallbacks
     def host_labels(self):
 
-        def _collect(dc, node):
-            host_inventory = {}
-            for folder in node.childEntity:
-                if hasattr(folder, 'host'):
-                    for host in folder.host:  # Iterate through Hosts in the Cluster
-                        host_name = host.summary.config.name.rstrip('.')
-                        host_inventory[host._moId] = [
-                            host_name,
-                            dc.name,
-                            folder.name if isinstance(folder, vim.ClusterComputeResource) else ''
-                        ]
-
-                if isinstance(folder, vim.Folder):
-                    host_inventory.update(_collect(dc, folder))
-
-            return host_inventory
+        def _collect(node, level=1, dc=None, folder=None):
+            inventory = {}
+            if isinstance(node, vim.Folder) and not isinstance(node, vim.StoragePod):
+                logging.debug(f"[F ] {('-' * level).ljust(7)} {node.name}")
```

Comment:
Don't Hate but for the [F ], CR H, DC. Can you add something a bit more descriptive? I hate having to bust out decoder rings.

Reply:
Changed this in the most up to date version and made flake8 happy.

I'll have to look into travis for the failing build. Trying to simulate it locally